tests/search_users_test.php

```diff
@@ -0,0 +1,130 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Search users
+ *
+ * @package    tool_mergeusers
+ * @copyright  2024 Leon Stringer <leon.stringer@ntlworld.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_mergeusers;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once(__DIR__ . '/../lib/mergeusersearch.php');
+
+/**
+ * Tests for searching for users.
+ */
+final class search_users_test extends \advanced_testcase {
+    /**
+     * Test search criteria works with PostgreSQL because this will generate an
+     * error when comparing and integer column (mdl_user.id) with a string.
+     * @dataProvider search_criteria
+     */
+    public function test_pgsqlsearch($searchfield, $input, $count): void {
```

Comment:
rename the test name, please, as you had before in the other PR.

Reply:
There's now only the `test_nodeletedusers` test.